pypeit/calibrations.py

```diff
@@ -915,23 +976,225 @@ def __repr__(self):
+                              'tilt': [buildimage.TiltImage, wavetilts.WaveTilts]
+                             }
+
+        # Get the name of the detector/mosaic
+        detname = spectrograph.get_det_name(det)
+
+        # Find the unique configuations in the metaddata
+        asn = {}
+        setups = fitstbl.unique_configurations(copy=True, rm_none=True)
+        if setup not in setups:
+            return asn
+
+        # Subset to output
+        if subset is None:
+            subset = np.ones(len(fitstbl), dtype=bool)
+
+        in_setup = fitstbl.find_configuration(setup) & subset
+        if not any(in_setup):
+            return asn
+
+        # Find all the frames in this calibration group
+        in_grp = fitstbl.find_calib_group(calib_ID) & in_setup
+        if not any(in_grp):
+            return asn
+
+        # Iterate through each frame type
+        for frametype, calib_classes in frame_calibrations.items():
+            indx = fitstbl.find_frames(frametype) & in_grp
+            if not any(indx):
+                continue
+            if not all(fitstbl['calib'][indx] == fitstbl['calib'][indx][0]):
+                msgs.error(f'CODING ERROR: All {frametype} frames in group {calib_ID} '
+                           'are not all associated with the same subset of calibration '
+                           'groups; calib for the first file is '
+                           f'{fitstbl["calib"][indx][0]}.')
+            calib_key = CalibFrame.construct_calib_key(setup, fitstbl['calib'][indx][0], detname)
+            asn[frametype] = {}
+            asn[frametype]['raw'] = fitstbl.frame_paths(indx)
+            asn[frametype]['proc'] \
+                    = [str(calib_class.construct_file_name(calib_key, calib_dir=_caldir))
+                            for calib_class in frame_calibrations[frametype]]
+            if must_exist:
+                asn[frametype]['proc'] \
+                    = [file for file in asn[frametype]['proc'] if Path(file).exists()]
+
+        if proc_only:
+            files = {}
+            for key, val in asn.items():
+                if not isinstance(val, dict) or 'proc' not in val:
+                    continue
+                for file in val['proc']:
+                    _file = Path(file).resolve()
+                    calib_type = _file.name.split('_')[0].upper()
+                    files['DIR'] = str(_file.parent)
+                    files[calib_type] = _file.name
+            return files
+
+        if include_science:
+            # Add any science and standard (and sky?) frames
+            for frametype in ['science', 'standard']:
+                indx = fitstbl.find_frames(frametype) & in_grp
+                if not any(indx):
+                    continue
+                asn[frametype] = fitstbl.frame_paths(indx)
+
+        return asn
+
+    @staticmethod
+    def association_summary(ofile, fitstbl, spectrograph, caldir, subset=None, det=None,
```

Comment:
as above, please insure there is  test on this new method.

Reply:
Yep, see `test_calibrations.py`.